core/cli/src/plugin/options.ts

```diff
@@ -74,48 +74,65 @@ export const substituteOptionTags = (plugin: Plugin, config: ValidPluginsConfig)
 
   // throw an error if there are tags in plugin option fields to avoid circular
   // references
-  const validateTagPath = (path: (string | number)[]) => {
+  const validateTagPath = (path: (string | number)[]): Validated<void> => {
```

Comment:
**suggestion**: to avoid the casting to `Validated<any>`, this could just return `boolean`. `substituted.push(invalid(['whatever']))` will infer the type correctly